### DISCUSSION TITLE
IPv6 prefix lifetime less than stated in RFC4862 5.5.3.e.2

### DISCUSSION THREAD

**Discussion by perkelix** [rsmarples/dhcpcd#116] [Q&A] [answered]
I have enabled PD from WAN to LAN.

It works well, except that the lifetime **dhcpcd** configures for the WAN interface is the same for valid_lft and preferred_lft, and the value (3600) is less than the minimum 2 hours stated in RFC4862 5.5.3.e.2, which Android devices abide to. This results in Android devices refusing to bind to a new prefix and drop their IPv6 connection.

Has anybody else encountered this? How did you solve it? Thanks!

**Comment by rsmarples** [comment, marked as answer]
dhcpcd gained RFC4862 5.5.3.e.2 support in 41b8ff0929f1c7ca9a270da24209a38271863e67.
I have clarified dhcpcd sending T1, T2 and IA vltime and pltimes in 8f86349e4919bb323342fff34f00bb1c2eb60d26.

Basically, the client SHOULD not send any T1, T2 or IA vltime and pltimes other than zero. The server MUST ignore any values sent in these fields.

So, you have two ways to fix this in a RFC compliant manner:
  1. Request the vendor of your Android devices to actually respect RFC 4862 properly as vltimes of less than two hours are perfectly valid.
  2. Request your upstream DHCPv6 allocate a vltime of two hours or greater.

As you're using Prefix Delegation, you do have a third option which is to advertise your prefix allocation to your downstream devices with whatever vltime and pltimes you see fit. How you achieve this is outside the scope of dhcpcd as it doens't do any advertising of prefixes.

**Comment by perkelix** [comment]
8f86349 makes perfect sense, since DHCPv6 doesn't request a lease. It uses whatever RA gave it.

Agreed that fixing Android's broken IPv6 implementation is outside the scope.

I guess that 9.4.2 is a wrap? :)
